### PR TITLE
Event listener

### PR DESCRIPTION
**Pull request by iMattPro**

### CODE REVIEW

File: event/listener.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	 */
+	public function show_collapsible_categories($event)
+	{
+		if (!isset($this->categories))
+		{
+			$this->categories = $this->operator->get_user_categories();
+		}
+
+		$cat_row = $event['cat_row'];
+		$cat_row += array('S_FORUM_HIDDEN' => in_array('fid_' . $event['row']['forum_id'], $this->categories));
```
Comment (rxu) [inline event/listener.php:91]: maybe rename `S_FORUM_HIDDEN` to something more ext-specific, like `S_CAT_COLLAPSED`?

File: tests/event/listener_test.php [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	 */
+	public function test_init_collapsible_categories()
+	{
+		$this->set_listener();
+
+		// Set up expectations
+		$this->template->expects($this->once())
+			->method('assign_vars')
+			->with(array(
+				'UA_COLLAPSIBLE_CATEGORIES_URL' => 'phpbb_collapsiblecategories_main_controller#' . serialize(array()),
```
Comment (rxu) [inline tests/event/listener_test.php:96]: We were going to use `json_encode()` or is `serialize()` just for tests?

Comment (iMattPro) [inline tests/event/listener_test.php:96]: @rxu this is something different and only related to this test of the controller helper route test stub

File: event/listener.php (hunk 1 above)
Comment (iMattPro) [inline event/listener.php:91]: Well it is based on this template event https://github.com/phpbb-extensions/collapsible-categories/blob/master/styles/all/template/event/forumlist_body_category_header_before.html

Everything else is `forumrow`, `FORUM_ID` so I think `FORUM_HIDDEN` seems a little more consistent.

**Pull request merged**